python-avd/pyavd/_eos_designs/eos_designs_facts/uplinks.py

```diff
@@ -436,7 +436,9 @@ def uplink_switch_interfaces(self: EosDesignsFactsGeneratorProtocol) -> EosDesig
             else:
                 msg = (
                     f"'uplink_switch_interfaces' is not set on '{self.shared_utils.hostname}' and 'uplink_switch' '{uplink_switch}' "
-                    f"does not have 'downlink_interfaces[{downlink_index}]' set under 'default_interfaces'."
+                    f"does not have 'downlink_interfaces[{downlink_index}]' set under 'default_interfaces' "
+                    f"(only {len(uplink_switch_facts._default_downlink_interfaces)} allocated). "
+                    f"This switch requires at least {downlink_index + 1} downlink_interfaces."
```

Comment:
please check the second comment from Paul in the issue. The idea would also be to compute exactly how many downlink interfaces are required (which would require more computation as we are inside the for loop)

Reply:
we will postpone this to #5773